message.go

```diff
@@ -2,26 +2,75 @@ package kafka
 
 import (
 	"encoding/json"
-	"fmt"
 
+	"github.com/blugnu/kafka/message"
 	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
 )
 
 var jsonUnmarshal = json.Unmarshal
 
-func MessageHeader(msg *Message, key string) string {
-	for _, h := range msg.Headers {
-		if string(h.Key) == key {
-			return string(h.Value)
+type MessageOption func(*Message) error
+
+// CreateMessage creates a new message with the specified topic and options. Any error
+// that occurs while creating the message is returned with a zero-value message.
+func CreateMessage(opts ...MessageOption) (Message, error) {
+	msg := kafka.Message{
+		TopicPartition: kafka.TopicPartition{
+			Partition: kafka.PartitionAny,
+		},
+	}
+	for _, opt := range opts {
+		if err := opt(&msg); err != nil {
+			return Message{}, err
 		}
 	}
-	return ""
+	return msg, nil
 }
 
-func OffsetString(tp kafka.TopicPartition) string {
-	return fmt.Sprintf("%s/%d:%d", *tp.Topic, tp.Partition, tp.Offset)
+// NewMessage returns a new kafka.Message configured with a specified topic and any of
+// the supported options, as required.
+//
+// # Errors
+//
+// If an error occurs while creating the message the function will panic.  If the caller
+// wishes to handle the error, they should use CreateMessage instead.
+//
+// # Parameters
+//
+// The function is generic with a type parameter T that specifies the type of the topic;
+// the topic type must support convertion to a string using fmt.Sprintf("%v").
+//
+// # Options
+//
+// Message options are applied in the order they are provided. Available options are:
+//
+//	message.Copy(msg *Message)            // copies the Key and Value from a specified message
+//	message.Header(key, value string)     // adds a header to the message
+//	message.Headers(map[string]string)    // adds headers to the message
+//	message.Key(key []byte)               // sets the message key
+//	message.Partition(partition int32)    // sets the message partition (will use kafka.AnyPartition by default)
+//	message.Topic[T](topic T)             // sets the message topic
+//	message.JSON(value any)               // sets the message value as a JSON representation of the given value
+//	message.String(value string)          // sets the message value to the specified string
+//	message.Value(value []byte)           // sets the message value to a copy of the specified byte slice
+//
+// # Example
+//
+//	msg := NewMessage("topic",
+//		message.Header("key", "value"),
+//		message.JSON(model),
+//	)
+func NewMessage[T comparable](topic T, opts ...MessageOption) Message {
+	opts = append([]MessageOption{message.Topic(topic)}, opts...)
+	msg, err := CreateMessage(opts...)
+	if err != nil {
+		panic(err)
+	}
+	return msg
```

Comment:
NewMessage deliberately panics since the errors that could occur would be programming errors.  If the caller does wish or need to handle any errors they would use the CreateMessage function instead.